pkg/stub/arbiter.go

```diff
@@ -0,0 +1,122 @@
+package stub
+
+import (
+	"fmt"
+	"github.com/Percona-Lab/percona-server-mongodb-operator/internal/mongod"
+	"github.com/Percona-Lab/percona-server-mongodb-operator/internal/util"
+	"github.com/Percona-Lab/percona-server-mongodb-operator/pkg/apis/psmdb/v1alpha1"
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (h *Handler) ensureReplsetArbiter(m *v1alpha1.PerconaServerMongoDB, replset *v1alpha1.ReplsetSpec, resources corev1.ResourceRequirements) (*appsv1.StatefulSet, error) {
+	if replset.Arbiter == nil {
+		replset.Arbiter = &v1alpha1.Arbiter{
+			Enabled: false,
+			Size:    0,
+		}
+	}
+	if replset.Arbiter.Enabled && replset.Arbiter.Size == 0 {
+		replset.Arbiter.Size = 1
+	}
+
+	arbiter := util.NewStatefulSet(m, m.Name+"-"+replset.Name+"arbiter")
+	if err := h.client.Get(arbiter); err != nil {
+		lf := logrus.Fields{
+			"version": m.Spec.Version,
+			"size":    replset.Size,
+			"cpu":     replset.Limits.Cpu,
+			"memory":  replset.Limits.Memory,
+			"storage": replset.Limits.Storage,
+		}
+		if replset.StorageClass != "" {
+			lf["storageClass"] = replset.StorageClass
+		}
+
+		arbiter, err := h.newArbiter(m, replset, resources)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := h.client.Create(arbiter); err != nil {
+			if !k8serrors.IsAlreadyExists(err) {
+				return nil, err
+			}
+		} else {
+			logrus.WithFields(lf).Infof("created arbiter stateful set for replset: %s", replset.Name)
+		}
+	}
+
+	return h.handleArbiterUpdate(m, arbiter, replset, resources)
+}
+
+func (h *Handler) handleArbiterUpdate(m *v1alpha1.PerconaServerMongoDB, arbiter *appsv1.StatefulSet, replset *v1alpha1.ReplsetSpec, resources corev1.ResourceRequirements) (*appsv1.StatefulSet, error) {
+	if arbiter.Spec.Replicas != nil && *arbiter.Spec.Replicas != replset.Arbiter.Size {
```

Comment:
@heartwilltell We will need to check that the arbiter count isn't too large here as well.

Reply:
Check, please. I added arbiterRightsizing(replset)